Cards/CardBase.gd

```diff
@@ -39,6 +41,11 @@ func _on_card_play(caster: Entity, target: Entity) -> void:
 	# TODO add other functionality that lots of cards may share (eg: restore AP, draw cards)
 
 
+func on_post_card_played():
```

Comment:
what is the goal of this ?

Reply:
This is so a card can draw or discard cards. This will change if we implement the CardEffect system mentioned in #14

Reply:
So for example, say we're at our hand limit of 10 cards. We play a "Draw 2 cards" card. What should happen is the card should discard and then we draw a new card. Then we will hit the hand limit of 10 cards. Without the `on_post_card_played`, the card we played won't be discarded yet. So we try to draw a new card before the queued card has been discarded. It will see that the hand is already full, and not draw the card.

Reply:
Now that I think about it, I think we can just rearrange the logic a bit so we only need one `_on_card_play()` that fires after the card discards. We just need to add a new `can_play_card()` that we check in the Battler first

Reply:
We can just have the card be discarded before its effects are played maybe ?

Reply:
Yeah, my latest commit does this

Reply:
Which commit are you talking about btw ?

Reply:
I see that we don't `_on_card_play(caster, target)` inside `try_play_card`, but I don't see where we play the card after discard

Reply:
ah nvm it's in Battler.gd, all good